src/app/view/OsReceive/OsReceiveProvider.tsx

```diff
@@ -62,7 +62,7 @@ export const OsReceiveProvider = ({
   }, [client?.isLogged])
 
   useEffect(() => {
-    if (!client || didCall.current) return
+    if (!client || didCall.current || state.filesToUpload.length === 0) return
 
     const fetchRegistry = async (): Promise<void> => {
```

Comment:
nit: We do not call the registry. We only call the cozy to fetch the apps. The method name should reflect that and should not be misleading https://github.com/cozy/cozy-flagship-app/blob/a9ff7f47579260290505bd62e617424c4f550d9e/src/app/domain/osReceive/models/OsReceiveCozyApp.ts#L58

Reply:
That's true, updating and then merging